File: src/erl/ifelse.erl
-module(ifelse).
-export([start/0]).

getValue() -> 3.

start() -> 
    put(value, true),

    First = get(value),

    if
        First -> io:fwrite("Got true~n");
        true -> io:fwrite("Got else~n")
    end,

    put(value, false),
    Second = get(value),
    if
        Second -> io:fwrite("Got true~n");
        true -> io:fwrite("Got else~n")
    end,

    Result = getValue(),
    if
        Result == 1 -> io:fwrite("Got true");
        Result == 2 -> io:fwrite("Got else");
        true -> io:fwrite("Got if else")
    end,

    erlang:halt().

File: src/ml/ifelse.ml
let value = ref true;;

if !value
then print_string "Got true\n"
else print_string "Got else\n";;

value := false;;
if !value
then print_string "Got true\n"
else print_string "Got else\n";;

let result = 3;;

if result = 1
then print_string "Got true\n"
else if result = 2
then print_string "Got else\n"
else print_string "Got if else\n";;